mirgecom/simutil.py

```diff
@@ -0,0 +1,111 @@
+__copyright__ = """
+Copyright (C) 2020 University of Illinois Board of Trustees
+"""
+
+__license__ = """
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+"""
+import numpy as np
+from meshmode.dof_array import thaw
+from mirgecom.io import (
+    make_io_fields,
+    make_status_message,
+    make_output_dump,
+)
+from mirgecom.checkstate import compare_states
+from mirgecom.euler import (
+    get_inviscid_timestep,
+)
+
+r"""
+This module provides some convenient utilities for
+building simulation applications.
+"""
+
+
+def check_step(step, n):
+    if n == 0:
+        return True
+    elif n < 0:
+        return False
+    elif step % n == 0:
+        return True
+    return False
+
+
+def inviscid_sim_timestep(discr, state, t, dt, cfl, eos,
+                              t_final, constant_cfl=False):
```

Comment:
Wouldn't this make more sense in the Euler module?

Reply:
Maybe.  I don't actually know.  It is a bit grey.. because on one hand, the user (i.e. the driver developer) should have control over what to do with the maximum stable dt returned by the Euler API.  In this function, I've created one possible (and approximately reasonable) thing the user could do with it.   Since it has one foot inside Euler and one foot in user land, I placed it here.